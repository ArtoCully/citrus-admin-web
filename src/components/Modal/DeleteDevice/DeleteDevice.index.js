// @flow
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { removeDevice } from 'src/redux/modules/device'
import DeleteDevice from './DeleteDevice'

type StateProps = {}

function mapStateToProps (
  state: GlobalReducerState,
  ownProps: Object,
): StateProps {

  return {}

}

type DispatchProps = {
  deleteDevice: Function,
}

function mapDispatchToProps (dispatch: any): DispatchProps {

  return {
    deleteDevice: bindActionCreators(removeDevice, dispatch),
  }

}

export default connect(mapStateToProps, mapDispatchToProps)(DeleteDevice)
export type ReduxProps = StateProps
