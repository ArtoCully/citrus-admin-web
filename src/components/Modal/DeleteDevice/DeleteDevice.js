// @flow
import React, { PureComponent } from 'react'
import LinkReplace from 'src/components/shared/LinkReplace/LinkReplace'
import css from './DeleteDevice.style.css'
import type { ReduxProps } from './DeleteDevice.index'


type Props = ReduxProps & {
  location: Object,
}

type State = {
  location: {
    device: Object
  },
}

class DeleteDevice extends PureComponent {

  state: State

  delete = (id: Number) => {

    this.props.deleteDevice(id)

  }

  props: Props

  render (): ?React$Element<any> {

    const {
      location: {
        state: {
          device = {},
        },
        pathname,
      },
    } = this.props

    return (
      <div className={css.deleteDevice}>
        <div className={css.header}>
          <div className={css.title}>DELETE DEVICE</div>
          <div className="pt4 pb4">
            Are you sure you want to delete {device.name}?
          </div>
        </div>
        <LinkReplace to={pathname}>
          <button className={css.btnCancel}>Cancel</button>
        </LinkReplace>
        <LinkReplace to={pathname}>
          <button
            className={`${css.btnDelete} ml2`}
            onClick={() => {

              this.delete(device.id)

            }}
          >
            Delete
          </button>
        </LinkReplace>
      </div>
    )

  }

}

export default DeleteDevice
