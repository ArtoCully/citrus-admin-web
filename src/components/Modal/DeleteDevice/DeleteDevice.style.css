.deleteDevice {
  @mixin font-size 14;
  width: 540px;
  height: 262px;
  border-radius: 5px;
  background-color: $colorWhite;
  box-shadow: 0 6px 6px 0 rgba(0, 0, 0, 0.3);
  text-align: center;
  padding: 2rem 1rem;
}

.header {
  margin: 10px 0 20px;
}

.title {
  @mixin font-size 22;
  font-weight: 500;
  text-align: center;
  color: $colorDark;
}

.subTitle {
  @mixin font-size 13;
}

.btnCancel, .btnDelete {
  background-color: $colorTheme;
  color: $colorWhite;
  width: 220px;
  height: 50px;
  align-self: center;
  margin-top: auto;
}
