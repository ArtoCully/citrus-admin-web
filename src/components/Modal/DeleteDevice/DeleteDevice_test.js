// @flow
/* eslint-disable max-len */

// Enzyme docs:
// http://airbnb.io/enzyme/docs/api/index.html

import React from 'react'
import { shallow } from 'enzyme'
import DeleteDevice from './DeleteDevice'


const mockProps = {
  location: {
    pathname: '/',
    state: {
      device: {},
    },
  },
  removeDevice: () => {},
  fetchDevices: () => {},
}


it('<DeleteDevice> renders the DeleteDevice content', () => {

  const wrapper = shallow(
    <DeleteDevice
      {...mockProps}
    />
  )

  expect(wrapper).toMatchSnapshot()

})
