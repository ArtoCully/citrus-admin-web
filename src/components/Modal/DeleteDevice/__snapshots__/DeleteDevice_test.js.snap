// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteDevice> renders the DeleteDevice content 1`] = `
<div
  className="deleteDevice"
>
  <div
    className="header"
  >
    <div
      className="title"
    >
      DELETE ANNOUNCEMENT
    </div>
    <div
      className="pt4 pb4"
    >
      Are you sure you want to delete 
      ?
    </div>
  </div>
  <LinkReplace
    to="/"
  >
    <button
      className="btnCancel"
    >
      Cancel
    </button>
  </LinkReplace>
  <LinkReplace
    to="/"
  >
    <button
      className="btnDelete ml2"
      onClick={[Function]}
    >
      Delete
    </button>
  </LinkReplace>
</div>
`;
